File: frontend/src/types/vcard.ts
export interface VCardLink {
  label: string;
  url: string;
  icon?: string;
}

export interface VCardService {
  title: string;
  description?: string;
  icon?: string;
}

export interface VCardTestimonial {
  name: string;
  text: string;
  role?: string;
  avatar_url?: string;
}

export interface VCardDocument {
  name: string;
  url: string;
  file_type?: string;
}

export interface VCardData {
  id: number;
  user_id: number;
  tenant_id: number | null;
  slug: string;
  status: 'draft' | 'published' | 'archived';
  template_id: string;
  name: string;
  title: string | null;
  company: string | null;
  profile_image_url: string | null;
  logo_url: string | null;
  phone: string | null;
  email: string | null;
  whatsapp: string | null;
  address: string | null;
  website: string | null;
  linkedin: string | null;
  instagram: string | null;
  github: string | null;
  twitter: string | null;
  youtube: string | null;
  custom_links: VCardLink[];
  bio: string | null;
  services: VCardService[];
  gallery: string[];
  video_url: string | null;
  testimonials: VCardTestimonial[];
  documents: VCardDocument[];
  custom_colors: Record<string, string>;
  custom_fonts: Record<string, string>;
  view_count: number;
  click_count: number;
  share_count: number;
  created_at: string | null;
  updated_at: string | null;
}

export type VCardCreatePayload = Omit<
  VCardData,
  'id' | 'user_id' | 'tenant_id' | 'slug' | 'status' | 'view_count' | 'click_count' | 'share_count' | 'created_at' | 'updated_at'
>;

export type VCardUpdatePayload = Partial<VCardCreatePayload> & { status?: string };

export interface VCardPublicData {
  slug: string;
  template_id: string;
  name: string;
  title: string | null;
  company: string | null;
  profile_image_url: string | null;
  logo_url: string | null;
  phone: string | null;
  email: string | null;
  whatsapp: string | null;
  address: string | null;
  website: string | null;
  linkedin: string | null;
  instagram: string | null;
  github: string | null;
  twitter: string | null;
  youtube: string | null;
  custom_links: VCardLink[];
  bio: string | null;
  services: VCardService[];
  gallery: string[];
  video_url: string | null;
  testimonials: VCardTestimonial[];
  documents: VCardDocument[];
  custom_colors: Record<string, string>;
  custom_fonts: Record<string, string>;
}

export interface TemplateDefinition {
  id: string;
  name: string;
  category: string;
  description: string;
  thumbnail: string;
}

File: backend/app/schemas/vcard.py
"""VCard Pydantic schemas."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class VCardLink(BaseModel):
    label: str
    url: str
    icon: Optional[str] = None


class VCardService(BaseModel):
    title: str
    description: Optional[str] = None
    icon: Optional[str] = None


class VCardTestimonial(BaseModel):
    name: str
    text: str
    role: Optional[str] = None
    avatar_url: Optional[str] = None


class VCardDocument(BaseModel):
    name: str
    url: str
    file_type: Optional[str] = None


class VCardCreateRequest(BaseModel):
    template_id: str = "minimal-clean"
    name: str = Field(..., min_length=1, max_length=200)
    title: Optional[str] = None
    company: Optional[str] = None
    profile_image_url: Optional[str] = None
    logo_url: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    whatsapp: Optional[str] = None
    address: Optional[str] = None
    website: Optional[str] = None
    linkedin: Optional[str] = None
    instagram: Optional[str] = None
    github: Optional[str] = None
    twitter: Optional[str] = None
    youtube: Optional[str] = None
    custom_links: list[VCardLink] = []
    bio: Optional[str] = None
    services: list[VCardService] = []
    gallery: list[str] = []
    video_url: Optional[str] = None
    testimonials: list[VCardTestimonial] = []
    documents: list[VCardDocument] = []
    custom_colors: dict = {}
    custom_fonts: dict = {}


class VCardUpdateRequest(BaseModel):
    template_id: Optional[str] = None
    name: Optional[str] = None
    title: Optional[str] = None
    company: Optional[str] = None
    profile_image_url: Optional[str] = None
    logo_url: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    whatsapp: Optional[str] = None
    address: Optional[str] = None
    website: Optional[str] = None
    linkedin: Optional[str] = None
    instagram: Optional[str] = None
    github: Optional[str] = None
    twitter: Optional[str] = None
    youtube: Optional[str] = None
    custom_links: Optional[list[VCardLink]] = None
    bio: Optional[str] = None
    services: Optional[list[VCardService]] = None
    gallery: Optional[list[str]] = None
    video_url: Optional[str] = None
    testimonials: Optional[list[VCardTestimonial]] = None
    documents: Optional[list[VCardDocument]] = None
    custom_colors: Optional[dict] = None
    custom_fonts: Optional[dict] = None
    status: Optional[str] = None


class VCardResponse(BaseModel):
    id: int
    user_id: int
    tenant_id: Optional[int] = None
    slug: str
    status: str
    template_id: str
    name: str
    title: Optional[str] = None
    company: Optional[str] = None
    profile_image_url: Optional[str] = None
    logo_url: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    whatsapp: Optional[str] = None
    address: Optional[str] = None
    website: Optional[str] = None
    linkedin: Optional[str] = None
    instagram: Optional[str] = None
    github: Optional[str] = None
    twitter: Optional[str] = None
    youtube: Optional[str] = None
    custom_links: list[VCardLink] = []
    bio: Optional[str] = None
    services: list[VCardService] = []
    gallery: list[str] = []
    video_url: Optional[str] = None
    testimonials: list[VCardTestimonial] = []
    documents: list[VCardDocument] = []
    custom_colors: dict = {}
    custom_fonts: dict = {}
    view_count: int = 0
    click_count: int = 0
    share_count: int = 0
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class VCardPublicResponse(BaseModel):
    """Slimmed response for public viewing — no user_id or tracking internals."""

    slug: str
    template_id: str
    name: str
    title: Optional[str] = None
    company: Optional[str] = None
    profile_image_url: Optional[str] = None
    logo_url: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    whatsapp: Optional[str] = None
    address: Optional[str] = None
    website: Optional[str] = None
    linkedin: Optional[str] = None
    instagram: Optional[str] = None
    github: Optional[str] = None
    twitter: Optional[str] = None
    youtube: Optional[str] = None
    custom_links: list[VCardLink] = []
    bio: Optional[str] = None
    services: list[VCardService] = []
    gallery: list[str] = []
    video_url: Optional[str] = None
    testimonials: list[VCardTestimonial] = []
    documents: list[VCardDocument] = []
    custom_colors: dict = {}
    custom_fonts: dict = {}

    class Config:
        from_attributes = True
